core/src/main/java/org/elasticsearch/index/mapper/core/DateFieldMapper.java

```diff
@@ -394,17 +394,21 @@ public long parseToMilliseconds(Object value, boolean roundUp,
         @Override
         public FieldStats.Date stats(IndexReader reader) throws IOException {
             String field = name();
-            long size = PointValues.size(reader, field);
-            if (size == 0) {
-                return null;
+            long size;
+            try {
+                size = PointValues.size(reader, field);
+                if (size == 0) {
+                    return new FieldStats.Date(reader.maxDoc(), isSearchable(), isAggregatable(), dateTimeFormatter());
+                }
+            } catch (Exception e) {
```

Comment:
done in https://github.com/elastic/elasticsearch/pull/17980/commits/574493ccb05cd854c2ed3fcc6385bb361c7630d0